File: Dante1/Module1.10/Zadanie10.11/main.c
#include <stdio.h>
#include "vector_utils.h"

int main()
{
    int array[150];

    printf("Enter smth: ");

    int size = read_vector(array, 150, 0);
    if(size == -1 || size == -2)
    {
        printf("Incorrect input");
        return 1;
    }
    if(size < 1)
    {
        printf("Not enough data available");
        return 3;
    }
    if(size > 150)
    {
        size = 150;
    }

    while(getchar()!='\n');

    int positions_to_shift;
    printf("Enter again: ");

    int check1 = scanf("%i",&positions_to_shift);
    if(check1 != 1)
    {
        printf("Incorrect input");
        return 1;
    }

    

    int dir;
    printf("Enter: ");
    int check2 = scanf("%i",&dir);
    if(check2 != 1)
    {
        printf("Incorrect input");
        return 1;
    }
    if(dir != 1 && dir != 2)
    {
        printf("Incorrect input data");
        return 2;
    }

    int check3 = shift(array,size,positions_to_shift,dir);
    if(check3 != 0)
    {
        printf("Incorrect input");
        return 1;
    }

    display_vector(array, size);

    return 0;
}

File: Dante1/Module1.10/Zadanie10.11/main.rs
use std::io::Read;

mod vector_utils;

fn main()
{
    let mut array:[i32; 150] = [0; 150];
    println!("Enter: ");

    let mut size: i32 = vector_utils::read_vector(&mut array, 150, 0);
    if size == -1 || size < 1
    {
        panic!("Incorrect input");
    }
    if size > 150
    {
        size = 150;
    }

    let mut input: String = String::new();
    println!("Enter: ");

    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let pos: i32 = input.trim().parse().expect("Incorrect parsing");

    input.clear();
    println!("Enter: ");

    std::io::stdin().read_line(&mut input).expect("Incorrect input");
    let dir: i32 = input.trim().parse().expect("Incorrect parsing");

    if dir == 1
    {
        if vector_utils::shift(&mut array, size as usize, pos, vector_utils::Direction::RotateRight) != 0
        {
            panic!("Incorrect data!");
        }
    }
    else
    {
        if vector_utils::shift(&mut array, size as usize, pos, vector_utils::Direction::RotateLeft) != 0
        {
            panic!("Incorrect data!");
        }
    }


    vector_utils::display_vector(&array, size);
}
